pkg/csi/service/wcp/controller.go

```diff
@@ -1811,11 +1811,55 @@ func (c *controller) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshot
 	}
 	deleteSnapshotInternal := func() (*csi.DeleteSnapshotResponse, error) {
 		csiSnapshotID := req.GetSnapshotId()
-		err := common.DeleteSnapshotUtil(ctx, c.manager.VolumeManager, csiSnapshotID)
-		if err != nil {
-			return nil, logger.LogNewErrorCodef(log, codes.Internal,
-				"Failed to delete WCP snapshot %q. Error: %+v",
-				csiSnapshotID, err)
+		isStorageQuotaM2FSSEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx,
+			common.StorageQuotaM2)
+		if isStorageQuotaM2FSSEnabled {
+			volumeID, _, err := common.ParseCSISnapshotID(csiSnapshotID)
+			if err != nil {
+				return nil, err
+			}
+			cnsVolumeInfo, err := volumeInfoService.GetVolumeInfoForVolumeID(ctx, volumeID)
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to retrieve cnsVolumeInfo for volume: %s Error: %+v", volumeID, err)
+			}
+			cnsSnapshotInfo, err := common.DeleteSnapshotUtil(ctx, c.manager.VolumeManager, csiSnapshotID,
+				&cnsvolume.DeletesnapshotExtraParams{
+					StorageClassName:           cnsVolumeInfo.Spec.StorageClassName,
+					StoragePolicyID:            cnsVolumeInfo.Spec.StoragePolicyID,
+					Capacity:                   cnsVolumeInfo.Spec.Capacity,
+					Namespace:                  cnsVolumeInfo.Namespace,
+					IsStorageQuotaM2FSSEnabled: isStorageQuotaM2FSSEnabled,
+				})
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"Failed to delete WCP snapshot %q. Error: %+v",
+					csiSnapshotID, err)
+			}
+
+			cnsVolumeInfo, err = volumeInfoService.GetVolumeInfoForVolumeID(ctx, cnsSnapshotInfo.SourceVolumeID)
```

Comment:
Why is this retrieved again?

Reply:
thought to fetch latest copy of cnsvolumeinfo, but realised it may not required